src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -14,70 +14,99 @@
-        .setApplicationName(APPLICATION_NAME)
-        .build();
-  }
+  private static final CommentAnalysis mockedAnalysis = mock(CommentAnalysis.class);
+  private static final CommentThreadListResponse mockedExpectedList =
+      mock(CommentThreadListResponse.class);
+  private static final CommentThreadListResponse mockedEdgeList =
+      mock(CommentThreadListResponse.class);
+  private static final ArrayList<Double> SCORE_IN_RANGE =
+      new ArrayList<>(Arrays.asList(0.001, 0.002, 0.003, 0.005, -0.1, -0.2));
+  private static final ArrayList<Double> EDGE_SCORE =
+      new ArrayList<>(Arrays.asList(1.0, -1.0, 0.0));
+  private static final ArrayList<Double> SYMMETRIC_SCORE =
+      new ArrayList<>(Arrays.asList(0.5, 0.9, -0.5, -0.9));
+  private static final ArrayList<Double> ALL_OUSIDE_SCORE =
+      new ArrayList<>(Arrays.asList(-2.0, -3.0, 3.0, -100.2));
+  private static final ArrayList<Double> ONE_OUSIDE_SCORE =
+      new ArrayList<>(Arrays.asList(-2.0, -1.0, 0.0));
+  private static final Statistics NORMAL_STAT = new Statistics(SCORE_IN_RANGE);
+  private static final Statistics EDGE_STAT = new Statistics(EDGE_SCORE);
+  private static final Statistics SYMMETRIC_STAT = new Statistics(SYMMETRIC_SCORE);
+  private static final Statistics ALL_OUSIDE_STAT = new Statistics(ALL_OUSIDE_SCORE);
+  private static final Statistics ONE_OUSIDE_STAT = new Statistics(ONE_OUSIDE_SCORE);
 
   @Before
-  public void setUp() throws GeneralSecurityException, IOException {
-    YouTube youtubeService = getService();
-    YouTube.CommentThreads.List youtuberequest = youtubeService.commentThreads().list(SNIPPET);
-    youtuberesponse =
-        youtuberequest
-            .setKey(DEVELOPER_KEY)
-            .setVideoId(TEST_VIDEO_ID)
-            .setMaxResults(2L)
-            .setTextFormat(PLAINTEXT)
-            .execute();
-    analysis = new CommentAnalysis();
+  public void setUp() {
+    when(mockedAnalysis.computeOverallStats(mockedExpectedList)).thenReturn(NORMAL_STAT);
+    when(mockedAnalysis.computeOverallStats(mockedEdgeList)).thenReturn(EDGE_STAT);
   }
 
   @Test
   public void testSentimentAnalysisInRange() {
     // Test the Sentiment Analysis Score within range -1 to 1.
-    Statistics result = analysis.computeOverallStats(youtuberesponse);
-    Assert.assertTrue(Math.abs(result.getAverageScore()) <= 1);
+    Statistics normalStat = mockedAnalysis.computeOverallStats(mockedExpectedList);
+    Assert.assertTrue(Math.abs(normalStat.getAverageScore()) <= 1);
+    Statistics edgeStat = mockedAnalysis.computeOverallStats(mockedEdgeList);
+    Assert.assertTrue(Math.abs(edgeStat.getAverageScore()) <= 1);
   }
 
   @Test
   public void testCategorizationEdgeCases() {
-    // TODO: test for splitting different scores into intervals
+    // Test with mocked analysis interface
+    Assert.assertEquals(
+        mockedAnalysis
+            .computeOverallStats(mockedExpectedList)
```

Comment:
Since we're mocking out `mockedAnalysis` here we don't necessarily need to invoke it.

These assertion statements could also be represented as:

`assertThat(SYMMETRIC_STAT.getAggregateValues()....)`

Reply:
Updated